src/LogOtter.CosmosDb.ContainerMock/ContainerMockData/ContainerData.cs

```diff
@@ -118,8 +126,6 @@ public Task<Response> UpsertItem(string json, PartitionKey partitionKey, ItemReq
 
         partition[id] = newItem;
 
-        DataChanged?.Invoke(this, new DataChangedEventArgs(isUpdate ? Operation.Updated : Operation.Created, newItem.Json, _serializationHelper));
```

Comment:
Won't this break things if the event isn't fired anymore?

Reply:
This change keeps it all inband but removing the event dispatch from the semaphore means if you do any changes which add items to the same container it won't be locked in the current AddItems call. We had some problems with the out of band testing(i.e. having to wrap all checks on snapshots with WaiFor, which ended up making large test sets become flaky)

Reply:
Yeah, but it looks like the event only fires for Add now instead of Add/Update

Reply:
You're right, I'll push up another PR to fix it